redwood/src/decryption.rs

```diff
@@ -58,20 +60,25 @@ impl<'a> DecryptionHelper for Helper<'a> {
             .key()
             .clone();
 
-        // Decrypt the secret key with the specified passphrase.
-        let mut pair = key.decrypt_secret(self.passphrase)?.into_keypair()?;
+        for pkesk in pkesks {
+            if pkesk.recipient() == &KeyID::from(key.fingerprint()) {
```

Comment:
Just a slight note that this would prevent messages encrypted using [`--throw-keyids`](https://superuser.com/a/1493135) from decryption (as then the `pkesk.recipient().is_wildcard()` is `true`).

The more elaborate example in `decrypt-with.rs` file also has the same issue: https://gitlab.com/sequoia-pgp/sequoia/blob/main/openpgp/examples/decrypt-with.rs#L100

(Which may, nor may not be important here but I'd like to point out drawbacks anyway).

Reply:
TIL about `--throw-keyids`. In this case we are only decrypting messages generated by us (either through the web UI or SecureDrop Client) and we don't use that option. I'll add a comment noting this limitation.